trace_decoder/src/decoding.rs

```diff
@@ -618,6 +628,47 @@ fn process_txn_info(
     Ok(gen_inputs)
 }
 
+fn merge_batch_jumpdest_tables<'t, T>(jdts: T) -> Option<JumpDestTableWitness>
+where
+    T: Iterator<Item = &'t Option<JumpDestTableWitness>>,
+{
+    let mut merged_table = JumpDestTableWitness::default();
+
+    let mut max_batch_ctx = 0;
+    for jdt in jdts {
+        let tx_offset = max_batch_ctx;
+        // abort if any transaction in the batch came without RPC JumpDestTable.
+        if jdt.is_none() {
+            return None;
+        }
+        for (code_hash, ctx_tbl) in jdt.as_ref().unwrap().0.iter() {
+            for (ctx, jumpsdests) in ctx_tbl.0.iter() {
+                let batch_ctx = tx_offset + ctx;
+                max_batch_ctx = max(max_batch_ctx, batch_ctx);
+
+                merged_table
+                    .0
+                    .entry(*code_hash)
+                    .or_insert(ContextJumpDests::default());
+
+                merged_table
+                    .0
+                    .get_mut(code_hash)
+                    .unwrap()
+                    .0
+                    .entry(batch_ctx)
+                    .or_insert(jumpsdests.clone());
+
+                debug_assert!(merged_table.0.contains_key(code_hash));
+                debug_assert!(merged_table.0[code_hash].0.contains_key(&batch_ctx));
+            }
+        }
+        max_batch_ctx = tx_offset;
```

Comment:
This is wrong I believe, we shouldn't reset `max_batch_ctx` here but keep the max value observed among contexts for the current txn (and possibly what's breaking batching currently)

Reply:
Yes. I think this should just be removed as `max_batch_ctx` is maintained in the inner loop.

Reply:
This was correctly spotted.